docs/pages/database-access/guides/azure-postgres-mysql.mdx

```diff
@@ -216,10 +201,87 @@ achieve that:
 </TabItem>
 </Tabs>
 
-## Step 3/4. Create Azure database users
+## Step 3/5. Configure IAM permissions for Teleport
+
+### Create a custom role
+
+Teleport needs Azure IAM permissions to be able to discover and register MySQL and PostgreSQL databases.
+Create a role with assignable scope(s) that include all databases that Teleport should discover. For example:
+
+```json
+{
+    "properties": {
+        "roleName": "TeleportDiscovery",
+        "description": "Allows Teleport to discover MySQL and PostgreSQL databases",
+        "assignableScopes": [
+            "/subscriptions/11111111-2222-3333-4444-555555555555"
+        ],
+        "permissions": [
+            {
+                "actions": [
+                    "Microsoft.DBforMySQL/servers/read",
+                    "Microsoft.DBforPostgreSQL/servers/read"
+                ],
+                "notActions": [],
+                "dataActions": [],
+                "notDataActions": []
+            }
+        ]
+    }
+}
+```
+
+This role definition allows Teleport to discover MySQL and PostgreSQL databases, but Teleport only needs 
+permissions for the database types you have. The assignable scopes include a subscription, so
+the role can be assigned at any resource scope within that subscription, or assigned using the
+subscription scope itself.
+
+<Admonition type="not" title="Custom role assignable scope">
+Custom roles, unlike
+[Azure built-in roles](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles),
+can not have a root assignable scope. The highest assignable scope that
+can be used in a custom role is subscription scope. Using a management group scope is currently an Azure
+preview feature, and only allows for a single management group in the "assignableScopes" of a role 
+definition.
+See [Azure RBAC custom roles](https://docs.microsoft.com/en-us/azure/role-based-access-control/custom-roles) for
+more information.
+</Admonition>
+
+Go to the [Subscriptions](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBlade) page and select a subscription.
+
+Click on *Access control (IAM)* in the subscription and select *Add > Add custom role*:
+<Figure align="left">
+![IAM custom role](../../../img/database-access/guides/azure/add-custom-role@2x.png)
+</Figure>
+
+In the custom role creation page, click the *JSON* tab and click *Edit*, then paste the JSON example
+and replace the subscription in "assignableScopes" with your own subscription id:
+<Figure align="left">
+![Create JSON role](../../../img/database-access/guides/azure/create-role-from-json@2x.png)
+</Figure>
+
+### Create a role assignment for the Teleport Database Service principal.
```

Comment:
Does this step conflict with `Using app registrations` in Step 2?

Reply:
Oh, you mean that the screenshot app name doesn't match? I'll fix that

Reply:
No, I wasn't sure if role assignment works for both managed identities and app registrations. Just tried and it's good. Ignore the previous comment.

Reply:
Oh yeah, it works for both. I can clarify that though since it may be a question that users will wonder too.